### PR TITLE
Add ability to view DS Blocks

### PR DESCRIPTION
**Pull request by lucac-zilliqa**
This is a pull request for adding the feature to view DS Blocks. 
This pull request adds the ability to view the most recent DS Blocks, a list of all DSBlocks, a detailed view of a single DS Block as well as the ability to search for DSBlocks using the search bar.

### CODE REVIEW

File: src/execution/DSBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import BlockLink from "../components/BlockLink";
+import DecoratedAddressLink from "./components/DecoratedAddressLink";
+import { RuntimeContext } from "../useRuntime";
+import { useLatestBlockChainInfo } from "../useLatestBlock";
+import { dsBlockURL } from "../url";
+import { useBlockPageTitle } from "../useTitle";
+import { useDSBlockData } from "../useZilliqaHooks";
+import { pubKeyToAddr, zilliqaToOtterscanTimestamp } from "../utils/utils";
+
+// TODO: Figure out what we want to do with the previous Hash field
```
Comment (lucac-zilliqa) [inline src/execution/DSBlock.tsx:20]: I have just displayed all the information. For some reason the API request returns the previous Hash. I do not know how to use it yet and was wondering if once devex is deprecated could we alter the API call to return the hash of the ds block which was requested

File: src/useZilliqaHooks.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
-import { useState, useEffect, useMemo } from "react";
-import useSWR, { Fetcher } from "swr";
+// TODO: Once devex is completely depricated we can alter the zilliqa APIs so
+// values are returned in a format that is easier to handle in Otterscan such
```
Comment (lucac-zilliqa) [inline src/useZilliqaHooks.ts:2]: This is just a future idea once this version of Otterscan becomes Zilliqa's default block explorer

File: src/Header.tsx [hunk 3]
```diff
@@ -45,8 +45,8 @@ const Header: FC<HeaderProps> = ({sourcifyPresent}) => {
             <input
               className="w-full rounded-l border-t border-b border-l px-2 py-1 text-sm focus:outline-none"
               type="text"
-              size={60}
-              placeholder={`Type "/" to search by address / txn hash / block number${
+              size={80}
+              placeholder={`Type "/" to search by address / txn hash / block number / # ds block number${
```
Comment (lucac-zilliqa) [inline src/Header.tsx:49]: This new searching methods works by having the search string for ds blocks be in the format "#<ds block number>"

File: src/execution/DSBlock.tsx (hunk 1 above)
Comment (JamesHinshelwood) [reply src/execution/DSBlock.tsx:20]: Is the issue that the response doesn't include the hash of the current DS block?

File: src/useLatestBlock.ts [hunk 4]
```diff
@@ -77,3 +82,39 @@ export const useLatestBlockNumber = (provider?: JsonRpcProvider) => {
...
+ * that'll update and trigger a component reder as a side effect every
+ * the poll returns a different value
+ */
+
+export const useLatestBlockChainInfo = (zilliqa?: Zilliqa) 
+: BlockchainInfo | undefined => {
+  const [latestBlockChainInfo, setLatestBlockChainInfo] = useState<BlockchainInfo>();
+
+  useEffect(() => {
+    // TODO: Is this necessary to check whether the hook has been removed
```
[the remark this answers is not in the document]
Comment (JamesHinshelwood) [reply src/useLatestBlock.ts:97]: I don't really understand in what circumstances `isCancelled = true`?

File: src/useZilliqaHooks.ts (hunk 2 above)
Comment (JamesHinshelwood) [reply src/useZilliqaHooks.ts:2]: We can't ever change those APIs - Many more consumers depend on their behaviour than just Devex.

**Review by JamesHinshelwood — APPROVED** [review]
LGTM, nice :)

File: src/useZilliqaHooks.ts (hunk 2 above)
Comment (lucac-zilliqa) [reply src/useZilliqaHooks.ts:2]: I see. Maybe we could instead implement new APIs purpose built for this version of otterscan then.

File: src/execution/DSBlock.tsx (hunk 1 above)
Comment (lucac-zilliqa) [reply src/execution/DSBlock.tsx:20]: It was more that I have no idea how knowing the hash of the previous block is useful. The only use that comes to mind is when moving between DS block pages where the previous DS block hash is used to get the page for the previous DS block but then moving forward between DS blocks would not be possible

**Pull request merged**